extensions/amp-social-share/1.0/social-share-config.js

```diff
@@ -14,19 +14,23 @@
  * limitations under the License.
  */
 
-import {dict} from '../../../src/utils/object';
-
 /**
  * Get social share configurations by supported type.
- * @param  {string} type
- * @return {!SocialShareConfigDef}
+ * @param  {?string} type
+ * @return {?SocialShareConfigDef}
  */
 export function getSocialConfig(type) {
   return BUILTINS[type];
 }
 
 /**
- * @typedef {!Object<string, !JsonObject|string>}
+ * @typedef {{
+ *   shareEndpont:           string,
+ *   defaultParams:          Object<string, string>,
+ *   defaultColor:           string,
+ *   defaultBackgroundColor: string,
+ *   bindings:               ?Array<string>,
```

Comment:
This should actually be `(!Array<string>|undefined)`. I.e. it's either absent or has a real value. The `?` indicates `nullable`, but it's not the case here.

Reply:
Thanks, I've done a full audit and removed most `?` throughout the files.  I had confused `nullable` and `|undefined`.